src/gui/scrollAndZoomView/ViewZoomer.cpp

```diff
@@ -64,8 +64,19 @@ void ViewZoomer::wheelZoom(QWheelEvent *event)
 	event->accept(); // supress scrolling
 }
 
+void ViewZoomer::fitImage()
+{
+	mView->fitInView(mView->sceneRect(), Qt::KeepAspectRatio);
+}
+
 void ViewZoomer::setZoomValue(double value)
 {
+	// ZoomPicker's mFitImageButton() sets value to -1
+	if (value < 0) {
```

Comment:
You're right, this is a bit strange. When you think about it, the zoom value has not yet change but rather you want to send a notification from the zoom picker to the ViewZoomer and the ViewZoomer then does something that changes the zoom value and this new zoom value should be sent back to the zoom picker. So maybe a new signal for the zoom picker that the ViewZoomer listens to?

Reply:
That means going trough `AnnotationGeneralSettings`, over `AnnotationSettingsAdapter` and land in `AbstractSettingsProvider` to access `mZoomValueProvider` :D

*hold my beer*

Reply:
Done, added as additional commit instead of `--amend`ing.